core/payment-driver/zksync/src/driver.rs

```diff
@@ -329,7 +343,28 @@ Mind that to be eligible you have to run your app at least once on testnet -
             .into_iter()
             .map(|allocation| allocation.remaining_amount)
             .sum();
-        Ok(msg.amount <= (account_balance - total_allocated_amount))
+
+        // NOTE: `wallet::get_tx_fee` accepts an _recipient_ address which is unknown at the moment
+        // so the _sender_ address is provider. This might bias fee calculation, because transaction
+        // to new account is little more expensive.
+        let tx_fee_cost = wallet::get_tx_fee(&msg.address, network).await?;
+        let total_txs_cost = tx_fee_cost * &(*TRANSACTIONS_PER_ALLOCATION);
+        let allocation_surcharge =
+            BigDecimal::min((*MAX_ALLOCATION_SURCHARGE).clone(), total_txs_cost);
```

Comment:
Redundant parentheses in line 351 (weird that `cargo fmt` didn't catch this one).
Unnecessary clone in line 352.

```suggestion
        let total_txs_cost = tx_fee_cost * &*TRANSACTIONS_PER_ALLOCATION;
        let allocation_surcharge = (&*MAX_ALLOCATION_SURCHARGE).min(&total_txs_cost);
```